packages/manager/modules/sms/src/sms/batches/create/routing.js

```diff
@@ -0,0 +1,170 @@
+import { SMS_COMPOSE } from '../../sms/compose/telecom-sms-sms-compose.constant';
+import SMS_CONSTANTS from '../../telecom-sms.constant';
+
+import component from './telecom-sms-batches-create.component';
+
+import contactsChoiceComponent from '../../components/contacts-choice/contacts-choice.component';
+import receiversChoiceComponent from '../../components/receivers-choice/receivers-choice.component';
+import composeTipComponent from '../../components/sms-tips/compose/sms-tips-compose.component';
+import sizeTipComponent from '../../components/sms-tips/size/sms-tips-size.component';
+
+export default /* @ngInject */ ($stateProvider) => {
+  $stateProvider.state('sms.service.batches.create', {
+    url: '/create',
+    views: {
+      'smsView@sms.service': {
+        component: component.name,
+      },
+    },
+    resolve: {
+      cancelSelection: /* @ngInject */ ($state) => () =>
+        $state.go('sms.service.batches.create'),
+      computeBatchPromises: /* @ngInject */ ($q) => (createBatchPromises) =>
```

Comment:
I'm having trouble understanding the purpose of this resolve, the `$q.all` is quite generic and not a specific call depending on a context so it could be in the service/controller imo

Reply:
This is done in order to avoid injection of `$q` in the controller.

Reply:
Seems to be something like : `qAll: /* @ngInject */ ($q) => (all) => $q.all(all)`
What are the benefits to avoid to inject `$q` in an angular.js controller ?